content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/snips.sh

```diff
@@ -0,0 +1,650 @@
+EOF
+}
+
+snip_perform_tls_origination_with_an_egress_gateway_4() {
+kubectl apply -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: direct-cnn-through-egress-gateway
+spec:
+  hosts:
+  - edition.cnn.com
+  gateways:
+  - istio-egressgateway
+  - mesh
+  http:
+  - match:
+    - gateways:
+      - mesh
+      port: 80
+    route:
+    - destination:
+        host: istio-egressgateway.istio-system.svc.cluster.local
+        subset: cnn
+        port:
+          number: 80
+      weight: 100
+  - match:
+    - gateways:
+      - istio-egressgateway
+      port: 80
+    route:
+    - destination:
+        host: edition.cnn.com
+        port:
+          number: 443
+      weight: 100
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: originate-tls-for-edition-cnn-com
+spec:
+  host: edition.cnn.com
+  trafficPolicy:
+    loadBalancer:
+      simple: ROUND_ROBIN
+    portLevelSettings:
+    - port:
+        number: 443
+      tls:
+        mode: SIMPLE # initiates HTTPS for connections to edition.cnn.com
+EOF
+}
+
+snip_perform_tls_origination_with_an_egress_gateway_5() {
+kubectl exec "${SOURCE_POD}" -c sleep -- curl -sL -o /dev/null -D - http://edition.cnn.com/politics
+}
+
+! read -r -d '' snip_perform_tls_origination_with_an_egress_gateway_5_out <<\ENDSNIP
+HTTP/1.1 200 OK
+...
+content-length: 150793
+...
+ENDSNIP
+
+snip_perform_tls_origination_with_an_egress_gateway_6() {
+kubectl logs -l istio=egressgateway -c istio-proxy -n istio-system | tail
+}
+
+snip_cleanup_the_tls_origination_example_1() {
+kubectl delete gateway istio-egressgateway
+kubectl delete serviceentry cnn
+kubectl delete virtualservice direct-cnn-through-egress-gateway
+kubectl delete destinationrule originate-tls-for-edition-cnn-com
+kubectl delete destinationrule egressgateway-for-cnn
+}
+
+snip_generate_client_and_server_certificates_and_keys_1() {
+git clone https://github.com/nicholasjackson/mtls-go-example
+}
+
+snip_generate_client_and_server_certificates_and_keys_2() {
+cd mtls-go-example || exit
+}
+
+snip_generate_client_and_server_certificates_and_keys_3() {
+yes | ./generate.sh nginx.example.com password
+}
+
+snip_generate_client_and_server_certificates_and_keys_4() {
+mkdir ../nginx.example.com && mv 1_root 2_intermediate 3_application 4_client ../nginx.example.com
+}
+
+snip_generate_client_and_server_certificates_and_keys_5() {
+cd .. || exit
+}
+
+snip_deploy_a_mutual_tls_server_1() {
+kubectl create namespace mesh-external
+}
+
+snip_deploy_a_mutual_tls_server_2() {
+kubectl create -n mesh-external secret tls nginx-server-certs --key nginx.example.com/3_application/private/nginx.example.com.key.pem --cert nginx.example.com/3_application/certs/nginx.example.com.cert.pem
+kubectl create -n mesh-external secret generic nginx-ca-certs --from-file=nginx.example.com/2_intermediate/certs/ca-chain.cert.pem
+}
+
+snip_deploy_a_mutual_tls_server_3() {
+cat <<EOF > ./nginx.conf
```

Comment:
FYI, to fix the lint errors in the lines below, you need to escape the heredoc. Go back to the index.md file and change the `<<EOF` to `<<\EOF`.

Reply:
I already did this, not sure why this wasn't propagated here after make gen. I'll double check this